src/main/extension/extension_helper.cpp

```diff
@@ -5,6 +5,8 @@
 #include "duckdb/main/client_context.hpp"
 #include "duckdb/common/string_util.hpp"
 
+#ifdef LINK_EXTENSIONS
```

Comment:
Same with the reversal here, to disable linking we should define `DISABLE_LINK_EXTENSIONS`, not the other way around. That way when nothing is defined the default behavior is triggered.